src/include/btree.i

```diff
@@ -1072,33 +1072,53 @@ __wt_ref_info(
     if (addr == NULL) {
         *addrp = NULL;
         *sizep = 0;
-        if (typep != NULL)
-            *typep = 0;
+        if (is_leafp != NULL)
+            *is_leafp = false;
     } else if (__wt_off_page(page, addr)) {
         *addrp = addr->addr;
         *sizep = addr->size;
-        if (typep != NULL)
-            switch (addr->type) {
-            case WT_ADDR_INT:
-                *typep = WT_CELL_ADDR_INT;
-                break;
-            case WT_ADDR_LEAF:
-                *typep = WT_CELL_ADDR_LEAF;
-                break;
-            case WT_ADDR_LEAF_NO:
-                *typep = WT_CELL_ADDR_LEAF_NO;
-                break;
-            default:
-                *typep = 0;
-                break;
-            }
+        if (is_leafp != NULL)
+            *is_leafp = addr->type != WT_CELL_ADDR_INT;
     } else {
         __wt_cell_unpack(session, page, (WT_CELL *)addr, unpack);
         *addrp = unpack->data;
         *sizep = unpack->size;
-        if (typep != NULL)
-            *typep = unpack->type;
+
+        if (is_leafp != NULL)
+            *is_leafp = unpack->type != WT_CELL_ADDR_INT;
+    }
+}
+
+/*
+ * __wt_ref_info_lock --
+ *     Lock the WT_REF and return the addr/size and type triplet for a reference.
+ */
+static inline void
+__wt_ref_info_lock(
+  WT_SESSION_IMPL *session, WT_REF *ref, uint8_t *addr_buf, size_t *sizep, bool *is_leafp)
+{
+    uint32_t previous_state;
+    const uint8_t *addr;
+
+    /*
+     * The WT_REF address references either an on-page cell or in-memory structure, and eviction
+     * frees both. If our caller is already blocking eviction (either because the WT_REF is locked
+     * or there's a hazard pointer on the page), we do no locking here and return a pointer to the
+     * address. If the WT_REF is not locked, we lock here, and copy out the address.
+     */
+    for (;; __wt_yield()) {
+        previous_state = WT_REF_LOCKED;
```

Comment:
The `previous_state` should point to `ref->state`. Here it is directly assigned `WT_REF_LOCKED`

Reply:
Thanks! Fixed in 03cc99e0d.